lib/reversetunnel/transport.go

```diff
@@ -438,12 +437,19 @@ func (p *transport) reply(req *ssh.Request, ok bool, msg []byte) {
 	}
 }
 
-// directDial attempst to directly dial to the target host.
-func (p *transport) directDial(servers []string, serverID string) (net.Conn, error) {
-	var errors []error
+// directDial attempts to directly dial to the target host.
+func (p *transport) directDial(servers []string) (net.Conn, error) {
+	if len(servers) <= 0 {
+		return nil, trace.BadParameter("no servers to dial")
+	}
 
+	var errors []error
 	for _, addr := range servers {
-		conn, err := net.Dial("tcp", addr)
+		dialer := net.Dialer{
+			Timeout: apidefaults.DefaultDialTimeout,
```

Comment:
Should we relax the timeout on the client side of this? We're going to dial a single server in pretty much every case except for fringe scenarios in which a proxy has more than one auth server in its configuration file (rather than the address of a dynamic load balancer of some sorts), but even in that case the client should perhaps have some extra grace time.